### PR TITLE
Batch cloudwatch metrics upload to reduce cpu usage

### PR DESCRIPTION
**Pull request by sammaritan12**
Additionally did some auto-formatting in vscode

https://app.asana.com/0/1163919217444318/1199567924494792/f

### CODE REVIEW

**Review by dooferlad — CHANGES_REQUESTED** [review]
Thanks for the PR!

Please remove the non-functional changes, since these just increase diff noise and make it harder to discern what the important changes are when looking back through the git history. Other than that, it looks good.

**Comment by sammaritan12** [comment]
@dooferlad all good, I use format on save in my IDE and it seemed to be overzealous in the formatting this time :)

File: files/cloudwatch/metrics.json [hunk 1]
```diff
@@ -0,0 +1,122 @@
+[
+    {
+        "MetricName": "CPUUtilization",
+        "Dimensions": [
+            {
+                "Name": "InstanceName",
+                "Value": $INSTANCE_NAME
```
Comment (dooferlad) [inline files/cloudwatch/metrics.json:7]: Why instance name, but not ID?

File: files/cloudwatch/metrics.json [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        "Timestamp": $TIMESTAMP,
+        "Value": $LOAD,
+        "Unit": "None"
+    },
+    {
+        "MetricName": "MemoryUsed",
+        "Dimensions": [
+            {
+                "Name": "InstanceId",
+                "Value": $INSTANCE_ID
```
Comment (dooferlad) [inline files/cloudwatch/metrics.json:35]: Why instance ID, but not name?

**Review by dooferlad — COMMENTED** [review]
Just a couple of questions about why we sometimes have instance IDs, other times instance names, and other times both. I would think that, if possible, having names and IDs would be best.

File: files/cloudwatch/metrics.json (hunk 1 above)
Comment (sammaritan12) [reply files/cloudwatch/metrics.json:7]: This was just a migration from single calls to a batched call, the original code didn't have Instance ID, so I didn't include it. If you think it's reasonable to add I can definitely add it.

```sh
$AWS_EXECUTABLE --region=$REGION cloudwatch put-metric-data --namespace $NAMESPACE --metric-name "CPUUtilization" --unit "Percent" --value $CPU --dimensions "InstanceName=$INSTANCE_NAME" --timestamp $TIMESTAMP
```

File: files/cloudwatch/metrics.json (hunk 2 above)
Comment (sammaritan12) [reply files/cloudwatch/metrics.json:35]: As above, but with the following code

```sh
$AWS_EXECUTABLE --region=$REGION cloudwatch put-metric-data --namespace $NAMESPACE --metric-name "MemoryUsed" --unit "Megabytes" --value $MEMORY_USED --dimensions "InstanceId=$INSTANCE_ID" --timestamp $TIMESTAMP

```

**Review by dooferlad — APPROVED** [review]

**Comment by sammaritan12** [comment]
Am merging, only things changed is the shell substitution didn't work in the `jq` command so I had to put the filename explicitly. Additionally I'm passing in a file to the AWS CLI because for some reason part of the bash prompt ends up in the variable.

**Pull request merged**